File: java/src/main/java/io/fivexer/sdk/model/PublicWorkerIdentity.java
package io.fivexer.sdk.model;

/**
 * A worker's portal credential. Never carries the PIN — only {@code hasPin}, whether one is set.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class PublicWorkerIdentity {
    private String id;
    private String workerId;
    private String label;
    private String status;
    private String email;
    private Boolean hasPin;
    private String activatedAt;
    private String createdAt;
    private String revokedAt;

    public String getId() { return id; }
    public String getWorkerId() { return workerId; }
    public String getLabel() { return label; }
    public String getStatus() { return status; }
    public String getEmail() { return email; }
    public Boolean getHasPin() { return hasPin; }
    public String getActivatedAt() { return activatedAt; }
    public String getCreatedAt() { return createdAt; }
    public String getRevokedAt() { return revokedAt; }
}

File: php/src/Model/PublicWorkerIdentity.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A worker's portal credential. Never carries the PIN — only whether one is set.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class PublicWorkerIdentity
{
    public function __construct(
        public readonly string $id,
        public readonly string $workerId,
        public readonly string $label,
        /** 'active' | 'revoked' */
        public readonly string $status,
        public readonly ?string $email,
        public readonly bool $hasPin,
        public readonly ?string $activatedAt,
        /** ISO-8601 */
        public readonly string $createdAt,
        public readonly ?string $revokedAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            workerId: (string) ($data['workerId'] ?? ''),
            label: (string) ($data['label'] ?? ''),
            status: (string) ($data['status'] ?? ''),
            email: isset($data['email']) ? (string) $data['email'] : null,
            hasPin: (bool) ($data['hasPin'] ?? false),
            activatedAt: isset($data['activatedAt']) ? (string) $data['activatedAt'] : null,
            createdAt: (string) ($data['createdAt'] ?? ''),
            revokedAt: isset($data['revokedAt']) ? (string) $data['revokedAt'] : null,
        );
    }
}
